pvmismatch/pvmismatch_lib/pvmodule.py

```diff
@@ -484,18 +498,48 @@ def calcMod(self):
                     Isub, Vsub = self.pvconst.calcParallel(
                         Iparallel, Vparallel, Vparallel.max(), Vparallel.min()
                     )
-            bypassed = Vsub < self.Vbypass
-            Vsub[bypassed] = self.Vbypass
+                    
+            # apply bypass diodes depending on the configuration passed
+            if type(self.Vbypass) == list:
```

Comment:
Note, this fails if `Vbypass` is another sequence like a tuple, a set, or a NumPy array. Instead use [duck typing](https://en.wikipedia.org/wiki/Duck_typing):

```python
try:
    num_bypass = len(self.Vbypass)
except TypeError:
    # base case - Vbypass across every cell string
    bypassed = Vsub < self.Vbypass
    Vsub[bypassed] = self.Vbypass
else:
    # if only one value is passed in the list- assume only one 
    # bypass diode  across the PV module
    if len(self.Vbypass) == 1: 
        bypass_entire_module = True
        Vbypass_mod = self.Vbypass[0]
        # if more than 1 values are passed, apply them across  
        # the cell strings in ascending order
    elif len(self.cell_pos) == len(self.Vbypass):    
        if self.Vbypass[substr_idx] is None:
            bypassed = None
        else:
            bypassed = Vsub < self.Vbypass[substr_idx]
            Vsub[bypassed] = self.Vbypass[substr_idx]
    else:
        raise Exception("wrong number of bypass diode values passed : %d"%(len(self.Vbypass)))
```

IMO you should always [prefer duck typing to explicit type checking](http://www.voidspace.org.uk/python/articles/duck_typing.shtml#problem). [Type checking is an antipattern](http://blog.millermedeiros.com/type-check-is-a-code-smell/) not only because [type chekcing can't get inherited types](https://docs.quantifiedcode.com/python-anti-patterns/readability/do_not_compare_types_use_isinstance.html), but any similar type.